### PR TITLE
Introduce Pulse 📡

### PR DESCRIPTION
**Pull request by tokijh**
# Introduce Pulse 📡

> Renamed to `Pulse` from `Signal` because it conflicts with `RxCocoa.Signal`

`Pulse` has diff only when mutated
To explain in code, the results are as follows.
```swift
var messagePulse: Pulse<String?> = Pulse(wrappedValue: "Hello tokijh")
let oldMessagePulse: Pulse<String?> = message
message = "Hello tokijh"
oldMessagePulse != messagePulse // true
oldMessagePulse.value == messagePulse.value // true
```

Use when you want to receive an event only if the new value is assigned, even if it is the same value.
like `alertMessage` (See follows or [PulseTests.swift](https://github.com/ReactorKit/ReactorKit/blob/master/Tests/ReactorKitTests/PulseTests.swift))
```swift
// Reactor
private final class MyReactor: Reactor {
  struct State {
    @Pulse var alertMessage: String?
  }
  func mutate(action: Action) -> Observable<Mutation> {
    switch action {
    case let .alert(message):
      return Observable.just(Mutation.setAlertMessage(message))
    }
  }
  func reduce(state: State, mutation: Mutation) -> State {
    var newState = state
    switch mutation {
    case let .setAlertMessage(alertMessage):
      newState.alertMessage = alertMessage
    }
    return newState
  }
}
// View
reactor.pulse(\.$alertMessage)
  .compactMap { $0 } // filter nil
  .subscribe(onNext: { [weak self] (message: String) in
    self?.showAlert(message)
  })
  .disposed(by: disposeBag)
// Cases
reactor.action.onNext(.alert("Hello"))  // showAlert() is called with `Hello`
reactor.action.onNext(.alert("Hello"))  // showAlert() is called with `Hello`
reactor.action.onNext(.doSomeAction)    // showAlert() is not called
reactor.action.onNext(.alert("Hello"))  // showAlert() is called with `Hello`
reactor.action.onNext(.alert("tokijh")) // showAlert() is called with `tokijh`
reactor.action.onNext(.doSomeAction)    // showAlert() is not called
```

### CODE REVIEW

File: Sources/ReactorKit/Pulse.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      self.valueUpdatedCount = UInt.min
+    } else {
+      self.valueUpdatedCount += 1
+    }
+  }
+}
+
+extension Pulse: Equatable {
+  public static func == (lhs: Pulse<Value>, rhs: Pulse<Value>) -> Bool {
+    return lhs.valueUpdatedCount == rhs.valueUpdatedCount
```
Comment (OhKanghoon) [inline Sources/ReactorKit/Pulse.swift:42]:
```suggestion
    return lhs.valueUpdatedCount == rhs.valueUpdatedCount 
      && lhs.value == rhs.value
```

How about comparing the value together?
I think there will be a problem when comparing pulses.

Comment (tokijh) [reply Sources/ReactorKit/Pulse.swift]: Good idea! I fixed it :) @OhKanghoon

**Review by OhKanghoon — APPROVED** [review]
👍

File: README.md [hunk 2]
```diff
@@ -358,6 +369,65 @@ final class MyReactor: Reactor {
...
+    case let .setAlertMessage(alertMessage):
+      newState.alertMessage = alertMessage
+    }
+
+    return newState
+  }
+}
+
+// View
+reactor.state.pulse(\.$alertMessage)
```
Comment (jsryudev) [inline README.md:415]: I think `reactor.state.pulse(\.$alertMessage)` is a typo of `reactor.pulse(\.$alertMessage)`
Is there any other way to use it?

Comment (tokijh) [reply README.md:415]: Oops! Thank you:)

**Pull request merged**